src/ngraph/runtime/cpu/pass/cpu_assignment.cpp

```diff
@@ -498,8 +498,12 @@ namespace ngraph
                         auto op_annotations =
                             std::make_shared<ngraph::runtime::cpu::CPUOpAnnotations>();
                         op_annotations->set_mkldnn_op(true);
-                        std::map<size_t, size_t> oi_pairs = {{0, 0}};
-                        op_annotations->set_in_place_oi_pairs(oi_pairs);
+                        if (get_user_count(node->get_argument(0).get()) == 1)
```

Comment:
I think this is overly conservative. Multiple users do not preclude us from re-using the storage. It's up to liveness to detect the **last** use and re-use the storage in that case. I'll bug you tomorrow about this.